signer/src/api/new_block.rs

```diff
@@ -914,4 +928,53 @@ mod tests {
         assert_eq!(db.rotate_keys_transactions.len(), 1);
         assert!(db.rotate_keys_transactions.get(&txid).is_some());
     }
+
+    #[test_case(EVENT_OBSERVER_BODY_LIMIT, true; "event within limit")]
+    #[test_case(EVENT_OBSERVER_BODY_LIMIT+1, false; "event over limit")]
```

Comment:
style nit that rustfmt cannot see:
```suggestion
    #[test_case(EVENT_OBSERVER_BODY_LIMIT + 1, false; "event over limit")]
```